File: src/main/java/pl/yalgrin/playnite/simplesync/domain/Platform.java
package pl.yalgrin.playnite.simplesync.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Table("playnite_platform")
public class Platform extends AbstractObjectEntity {
    @Id
    @Column("id")
    private Long id;

    @Column("playnite_id")
    private String playniteId;

    @Column("name")
    private String name;

    @Column("specification_id")
    private String specificationId;

    @Column("icon_md5")
    private String iconMd5;

    @Column("cover_image_md5")
    private String coverImageMd5;

    @Column("background_image_md5")
    private String backgroundImageMd5;

    @Column("removed")
    private boolean removed;

    @Transient
    private boolean notifyAll;

    @Transient
    private boolean changed;
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/library/domain/Platform.kt
package pl.yalgrin.playnite.simplesync.library.domain

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Transient
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table

@Table("playnite_platform")
data class Platform(
    @Id
    @Column("id")
    override var id: Long? = null,

    @Column("playnite_id")
    override var playniteId: String? = null,

    @Column("name")
    override var name: String? = null,

    @Column("specification_id")
    var specificationId: String? = null,

    @Column("icon_md5")
    var iconMd5: String? = null,

    @Column("cover_image_md5")
    var coverImageMd5: String? = null,

    @Column("background_image_md5")
    var backgroundImageMd5: String? = null,

    @Column("removed")
    override var isRemoved: Boolean = false,

    @Transient
    override var isNotifyAll: Boolean = false,

    @Transient
    override var isChanged: Boolean = false
) : LibraryObjectEntity